### PR TITLE
Refactor/ensure not revoked

### PR DESCRIPTION
**Pull request by raiseandfall**
- combines `ensureNotRevokedByList` and `ensureNotRevokedBySpentOutput` into `ensureNotRevoked` for less code. Tests in place.

### CODE REVIEW

File: src/helpers/array.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function intersect (array1 = [], array2 = []) {
+  return array1.filter(value => array2.indexOf(value) !== -1).filter((e, i, c) => c.indexOf(e) === i);
```
Comment (lemoustachiste) [inline src/helpers/array.js]: maybe explicit what are `e`, `i` and `c` with more verbose names?

File: src/inspectors/ensureNotRevoked.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  if (!revokedAddresses || !keys) {
+    // nothing to do
+    return;
+  }
+
+  if (!Array.isArray(keys)) {
+    keys = [keys];
+  }
+
+  keys = keys.filter(key => key != null && typeof key !== 'undefined');
```
Comment (lemoustachiste) [inline src/inspectors/ensureNotRevoked.js]: I think `key != null` already checks for the undefined case.

File: test/application/inspectors/ensureHashesEqual.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    describe('given it is called with two similar hashes', function () {
+      it('should not throw an error', function () {
+        const hash = 'hash';
+        expect(function () {
+          ensureHashesEqual(hash, hash);
+        }).not.toThrow();
+      });
+    });
+
+    describe('given it is called with two different hashes', () => {
```
Comment (lemoustachiste) [inline test/application/inspectors/ensureHashesEqual.test.js]: arrow function

File: test/application/helpers/array.test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { intersect } from '../../../src/helpers/array';
+
+describe('Array Helper', () => {
+  describe('intersect method', () => {
+    describe('given it is called with two arrays', () => {
+      it('should return an array of intersect', () => {
```
Comment (lemoustachiste) [inline test/application/helpers/array.test.js]: arrow functions

File: src/helpers/array.js (hunk 1 above)
Comment (raiseandfall) [reply src/helpers/array.js]: Good one. I like to live on the edge a bit too much.

File: test/application/inspectors/ensureHashesEqual.test.js (hunk 3 above)
Comment (raiseandfall) [reply test/application/inspectors/ensureHashesEqual.test.js]: 👍

File: test/application/helpers/array.test.js (hunk 4 above)
Comment (raiseandfall) [reply test/application/helpers/array.test.js]: 👍

**Pull request merged**